lib/ResetPassword.dart

```diff
@@ -0,0 +1,203 @@
+    var size = MediaQuery.of(context).size.aspectRatio;
+    Future<void> successDialog() {
+      // flutter defined function
+      return showDialog(
+        context: context,
+        builder: (BuildContext context) {
+          // return object of type Dialog
+          return CupertinoAlertDialog(
+            title: new Text(
+              "Success!",
+              textAlign: TextAlign.center,
+            ),
+            content: new Text(
+              'Reset link has been sent to your email.',
+            ),
+            actions: <Widget>[
+              // usually buttons at the bottom of the dialog
+              new FlatButton(
+                child: new Text(
+                  "Close",
+                  style: TextStyle(
+                      color: Utils.header,
+                      fontWeight: FontWeight.w600,
+                      fontSize: 17),
+                  textAlign: TextAlign.center,
+                ),
+                onPressed: () {
+                  Navigator.of(context).pop();
+                  Navigator.of(context).pop();
+                },
+              ),
+            ],
+          );
+        },
+      );
+    }
+
+    void _showError() {
+      String errorMessage =
+          "No user found with that email, Please check your email again or try to register.";
+
+      showDialog(
+        context: context,
+        builder: (BuildContext context) {
+          return CupertinoAlertDialog(
+            title: Text("Oops!"),
+            content: new Text(errorMessage),
+            actions: <Widget>[
+              new FlatButton(
+                child: new Text(
+                  "Close",
+                  textAlign: TextAlign.center,
+                  style: TextStyle(
+                      color: Utils.header,
+                      fontWeight: FontWeight.w600,
+                      fontSize: 17),
+                ),
+                onPressed: () {
+                  Navigator.of(context).pop();
+                },
+              ),
+            ],
+          );
+        },
+      );
+    }
+
+    final resetTitle = Text.rich(
+      TextSpan(children: <TextSpan>[
+        TextSpan(
+            text: ' Reset your password ',
+            style: TextStyle(
+              fontWeight: FontWeight.bold,
+              color: Utils.header,
+              fontSize: size * 40,
+              fontFamily: "Roboto",
+            ))
+      ]),
+    );
+    Widget emailField = TextFormField(
+      controller: emController,
+      obscureText: false,
+      style: style,
+      decoration: InputDecoration(
+          contentPadding: EdgeInsets.fromLTRB(
+              width * 0.05, height * 0.025, width * 0.02, height * 0.02),
+          hintText: "Email ( @dell / @dellteams )",
+          hintStyle: TextStyle(fontSize: 16),
+          border:
+              OutlineInputBorder(borderRadius: BorderRadius.circular(15.0))),
+    );
+    Widget resetButton = Container(
```

Comment:
For future reference, it is usually better to create a new class for widgets (specially ones that call `setState()`) instead of using functions that return `Widget`  (using variables will have the same side effects as functions). 

See https://stackoverflow.com/questions/53234825/what-is-the-difference-between-functions-and-classes-to-create-reusable-widgets

Reply:
Very helpful thank you! @Ali-Amin